### ISSUE TITLE
Create a macro for linking to an OLID

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary#2283]
### Is your feature request related to a problem? Please describe.
Open Library supports macros (which are like "templates" in Wikipedia speak) in description fields. I.e. `{{Hello("Foo")}}` will trigger the `/macros/Hello.html` template, and render the text "Hello, Foo!". We should create a template, maybe `{{OLID('OL16551541W')}}` that creates a link to a record, automatically setting the link's title (avoiding duplicating data). This would also help with localization later on (since the title could be rendered differently for different locales).

### Describe the solution you'd like
- A way to link to a record without having to duplicate the record's name.

### Proposal & Constraints
- A macro called `OLID` which takes an olid and renders an `<a>` link with the record's name automatically populated.

### Additional context
@seabelis has had to create links like this a lot as she logs inter-work relationships. E.g. https://openlibrary.org/works/OL15841605W/The_Awakening

### Stakeholders
@seabelis

**Comment by seabelis** [comment]
Related to https://github.com/internetarchive/openlibrary/issues/1808

**Comment by seabelis** [comment]
Would a dropdown similar to the identifiers be possible. Some relationships that would be useful are:

Contained in: for works also collected elsewhere or partial works
Contains: for collections/anthologies
Preceded by: for ordered series
Followed by: for ordered series
Adaptation of: for original work
Adaptations of this work: for adapted works
Other works about this work: link to study guides, criticism
This work is about: links from study, guides, criticism, etc.

**Comment by cdrini** [comment]
That would be a better long-term solution for #1808 ; this would be something we could use in the meantime though (and would also be usable from like profile pages and other docs).

**Comment by b-thebest** [comment]
Can I work on this issue

**Comment by mekarpeles** [comment]
Hi @b2pacific and @b-thebest, I apologize for not replying sooner!

@b-thebest since you asked first, I'm adding you as the assignee. If you may no longer have the availability, let us know so I can offer @b2pacific a chance :)

thank you both!

**Comment by mekarpeles** [comment]
@b-thebest did you still wish to work on this one? Otherwise I'd like to give @b2pacific a chance

**Comment by b-thebest** [comment]
@mekarpeles I would like to work on this issue. I am starting to work on this.

Thanks

**Comment by JaydenTeoh** [comment]
@mekarpeles I know this issue is dated but do you still need a hand with it?

**Comment by mekarpeles** [comment]
@psyren99 sure, give it a try :)

**Comment by mekarpeles** [comment]
closing as non-issue, will merge in the corresponding PR if it works but its currently blocked on conflicts

**Comment by RayBB** [comment]
reopening so @mekarpeles can be notified the PR is waiting on him

**Comment by mekarpeles** [comment]
So the idea is to have a `{{BookMacro("OL123M")}}` that a librarian can use to render a cover, i18n'd title, possibly author (similar to google docs chip feature)

**Comment by RayBB** [comment]
@psyren99 perhaps you can create a new PR?
